pkg/loglines/dedup.go

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package loglines
+
+import (
+	"fmt"
+	"os"
+	"sort"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	keepGenerations = 5
+)
+
+func dedupGeneration(lineSlices []*LineSlice) *LineSlice {
+	ls1, ls2 := DedupLineSlices(lineSlices)
+	output := MakeSliceFromLines(MakeNewCombinedSlice(ls1.Lines, ls2.Lines), ls2.Generation)
+	return output
+}
+
+// findLineIndex will find the index of a line in a slice of lines
+// and will return -1 if it is not found
+func findLineIndex(needle *ProcessedLine, list []*ProcessedLine) int {
+	checkLine := needle.Full
+	for i, hay := range list {
+		if hay.Full == checkLine {
+			return i
+		}
+	}
+	return -1
+}
+
+//nolint:varnamelen // x and y are just two sets of lines
+func findIncompatableLines(x, y []*ProcessedLine) int {
```

Comment:
Fix typo in name and add a definition of what an incompatible line is as it's extensively used:

```suggestion
// an incompatible line is a line that exists in x without being present at the same index in y.
func findIncompatibleLines(x, y []*ProcessedLine) int {
```

Reply:
also consider renaming to something like `findFirstIssueIndex`